File: schedule/src/entity/course.ts
import {Schedule} from './schedule';

/**
 * 课程
 */
export class Course {
  id: number;
  name: string | undefined;
  hours: number | undefined;
  schedules: Schedule[] = [];
  constructor(data = {} as {
    id?: number
    name?: string,
    hours?: number,
    schedules?: Schedule[],
  }) {
    this.id = (data.id as number);
    this.name = data.name;
    this.hours = data.hours;
    this.schedules = data.schedules ? data.schedules : [];
  }
}

File: api/src/main/java/com/yunzhi/schedule/entity/Course.java
package com.yunzhi.schedule.entity;

import com.fasterxml.jackson.annotation.JsonView;
import io.swagger.annotations.ApiModelProperty;
import org.hibernate.annotations.SQLDelete;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
@SQLDelete(sql = "update `course` set deleted = 1 where id = ?")
public class Course implements SoftDelete {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @ApiModelProperty("id")
    @JsonView(IdJsonView.class)
    private Long id;

    @ApiModelProperty("名称")
    @JsonView(NameJsonView.class)
    private String name;

    @ApiModelProperty("课时")
    @JsonView(HoursJsonView.class)
    private String hours;

    @ApiModelProperty("是否已删除")
    private Boolean deleted = false;
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getHours() {
        return hours;
    }

    public void setHours(String hours) {
        this.hours = hours;
    }
    @Override
    public Boolean getDeleted() {
        return this.deleted;
    }

    private void setDeleted(Boolean deleted) {
        this.deleted = deleted;
    }

    public interface IdJsonView {}
    public interface NameJsonView {}
    public interface HoursJsonView {}

}
